### PR TITLE
Feat: Edition drop clone from thirdweb

### PR DESCRIPTION
**Pull request by viettu-bic**

### CODE REVIEW

File: test/edition/BicEdition.t.sol
Comment (ntnam-bic) [inline test/edition/BicEdition.t.sol:1]: add thêm unit test cho các edge cases liên quan đến soldout, quantityLimitPerWallet, etc.

File: script/edition/Edition20250902Script.s.sol
Comment (ntnam-bic) [inline script/edition/Edition20250902Script.s.sol:1]: remove _TEST với chuyển URI to env (sẽ merge vô master) với add thêm env cho số lượng premint trong batchMint

File: script/edition/Edition20250902Script.s.sol [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    function createDropConditions(
+        BicEdition bicEdition
+    ) internal {
+        address currency = vm.envAddress("BIC_ADDRESS_TESTNET");
+        IClaimCondition.ClaimCondition[] memory conditionsForId0 = new IClaimCondition.ClaimCondition[](1);
+        conditionsForId0[0] = IClaimCondition.ClaimCondition({
+            startTimestamp: block.timestamp,
+            maxClaimableSupply: 5000,
```
Comment (ntnam-bic) [inline script/edition/Edition20250902Script.s.sol:54]: update drop condition để align với req
